Add Overlay documentation (#76)
Linkurious/docker-lke: README.md

# Dockerized repository

## Configuration

### Templated configuration

Configure variables in .env. See [.env.example](.env.example) for template.
Reasonable defaults are provided.
We provide detailed documentation for [variable-expansion](https://doc.linkurious.com/admin-manual/latest/configure/#variable-expansion)

You will most likely need to set your private repository host, if images are not available locally.

This mode has limitations and will be dropped in the future.

### Configuration overlay

As an alternative, you may put a partial configuration file in your data folder: `data/config/production-overlay.json`.

> This file will be overlayed on top of your current production.json file. If no `production.json` file is provided, defaults will be generated.

Reviewer: overlaid?

Author: typo

> This file will be overlayed on top of your current production.json file. If no `production.json` file is provided, defaults will be generated.

Reviewer: maybe explain what this means:
- having these values in the config: the config value is ignored
- changing the config value (via config file or ui): the change is ignored, the value from the overlay continues to be used
- be careful about possibly creating inconsistent config: the merged result of config + overlay might not pass validation

Author: pushed an edit to match your proposals.

# Dockerized repository

## Configuration

### Templated configuration

Configure variables in .env. See [.env.example](.env.example) for template.
Reasonable defaults are provided.
We provide detailed documentation for [variable-expansion](https://doc.linkurious.com/admin-manual/latest/configure/#variable-expansion)

You will most likely need to set your private repository host, if images are not available locally.

This mode has limitations and will be dropped in the future.

### Configuration overlay

As an alternative, you may put a partial configuration file in your data folder: `data/config/production-overlay.json`.
This file will be overlaid on top of your current production.json file. If no `production.json` file is provided, defaults will be generated.

Overlay file will take precedence on anything included in the main configuration file.

> Values present in the overlay are no longer modifiable via the UI. Overlay file always takes precendenc.

Reviewer: ```suggestion
Values present in the overlay are no longer modifiable via the UI. Overlay file values always takes precedence.
```

> As the final configuration is the result of the merge of two config files, you can possibly create an inconsistent configuration file, that will not pass validation, and thus prevent Linkurious Enterprise to start.

Reviewer: ```suggestion
As the final configuration is the result of merging two files (overlay & configuration), it is possible to create an inconsistent configuration that will not pass validation, thus preventing Linkurious Enterprise from starting.
```
